File: v0-template/scripts/create-zip.mjs
import archiver from 'archiver';
import fs from 'fs';
import path from 'path';

const sourceDir = '/vercel/share/v0-project';
const outputFile = path.join(sourceDir, 'bullshark-project.zip');

const output = fs.createWriteStream(outputFile);
const archive = archiver('zip', { zlib: { level: 9 } });

output.on('close', () => {
  console.log(`[v0] Zip file created: ${outputFile}`);
  console.log(`[v0] Size: ${(archive.pointer() / 1024 / 1024).toFixed(2)} MB`);
});

archive.on('error', (err) => {
  console.error(`[v0] Archive error: ${err.message}`);
  process.exit(1);
});

archive.pipe(output);

// Add all files except node_modules, .next, .git, and logs
archive.glob('**/*', {
  cwd: sourceDir,
  ignore: ['node_modules/**', '.next/**', '.git/**', '*.log', 'bullshark-project.zip']
});

archive.finalize();

File: v0-template/scripts/create_zip.py
import os
import zipfile
from pathlib import Path

# Define project root and output path
project_root = Path("/vercel/share/v0-project")
output_zip = project_root / "bullshark-project.zip"

# Files and directories to exclude
exclude_dirs = {".next", "node_modules", ".git", ".env.local", "dist", "build"}
exclude_files = {".env", ".env.local", ".DS_Store"}

def should_exclude(path: Path, relative_to: Path) -> bool:
    """Check if a path should be excluded from the zip."""
    parts = path.relative_to(relative_to).parts
    
    # Check if any part of the path is in exclude_dirs
    for part in parts:
        if part in exclude_dirs:
            return True
    
    # Check if filename is in exclude_files
    if path.name in exclude_files:
        return True
    
    return False

# Create the zip file
with zipfile.ZipFile(output_zip, 'w', zipfile.ZIP_DEFLATED) as zipf:
    for root, dirs, files in os.walk(project_root):
        # Modify dirs in-place to skip excluded directories
        dirs[:] = [d for d in dirs if d not in exclude_dirs]
        
        for file in files:
            file_path = Path(root) / file
            
            if not should_exclude(file_path, project_root):
                arcname = file_path.relative_to(project_root)
                zipf.write(file_path, arcname)
                print(f"Added: {arcname}")

print(f"\n✓ Zip file created successfully: {output_zip}")
print(f"Size: {os.path.getsize(output_zip) / (1024*1024):.2f} MB")
